### PR TITLE
Admin: Support /eval in PMs

### PR DESCRIPTION
**Pull request by mia-pi-git**
QoL.

### CODE REVIEW

File: server/chat-commands/admin.ts [hunk 1]
```diff
@@ -1209,25 +1208,25 @@ export const commands: Chat.ChatCommands = {
...
-			const battle = room.battle;
+			const battle = room?.battle;
 			const me = user;
 			let result = eval(target);
 			/* eslint-enable no-eval, @typescript-eslint/no-unused-vars */
 
 			if (result?.then) {
-				uhtmlId = `eval-${room.nextGameNumber()}`;
-				this.sendReply(`|uhtml|${uhtmlId}|${generateHTML('<', 'Promise pending')}`);
+				uhtmlId = `eval-${room ? room.nextGameNumber() : Date.now()}`;
```
Comment (Zarel) [inline server/chat-commands/admin.ts:1217]: Honestly let's not do `nextGameNumber` at all. Just `Date.now()` + `Math.random()`.

Comment (Zarel) [reply server/chat-commands/admin.ts:1217]: Actually Date.now() is millisecond precision? nvm no need for random, either

```suggestion
				uhtmlId = `eval-${Date.now()}`;
```

File: server/chat-commands/admin.ts [hunk 2]
```diff
@@ -1209,25 +1208,25 @@ export const commands: Chat.ChatCommands = {
...
-			const battle = room.battle;
+			const battle = room?.battle;
 			const me = user;
 			let result = eval(target);
 			/* eslint-enable no-eval, @typescript-eslint/no-unused-vars */
 
 			if (result?.then) {
-				uhtmlId = `eval-${room.nextGameNumber()}`;
-				this.sendReply(`|uhtml|${uhtmlId}|${generateHTML('<', 'Promise pending')}`);
+				uhtmlId = `eval-${Date.now())}`;
```
Comment (Zarel) [inline server/chat-commands/admin.ts:1217]: Actually on second thought there _is_ a need for `Math.random`. You can send multiple evals in the same loop.

```suggestion
				uhtmlId = `eval-${Date.now().toString().slice(-6)}-${Math.random().toFixed(6).slice(-6)}`;
```

File: server/chat-commands/admin.ts [hunk 3]
```diff
@@ -1209,25 +1208,25 @@ export const commands: Chat.ChatCommands = {
...
+			const battle = room?.battle;
 			const me = user;
 			let result = eval(target);
 			/* eslint-enable no-eval, @typescript-eslint/no-unused-vars */
 
 			if (result?.then) {
-				uhtmlId = `eval-${room.nextGameNumber()}`;
-				this.sendReply(`|uhtml|${uhtmlId}|${generateHTML('<', 'Promise pending')}`);
+				uhtmlId = `eval-${Date.now().toString().slice(-6)}-${Math.random().toFixed(6).slice(-6)}`;
+				this.sendReply(`|c|~|/uhtml ${uhtmlId},${generateHTML('<', 'Promise pending')}`);
```
Comment (Zarel) [inline server/chat-commands/admin.ts:1218]: Possibly `sendReply` should auto-convert `|uhtml|`. It does already auto-convert `|html|`, right?

Comment (mia-pi-git) [reply server/chat-commands/admin.ts]: (done, now that it does do that)

**Comment by Zarel** [comment]
tyty

**Pull request merged**